rpmalloc/rpmalloc.c

```diff
@@ -1834,7 +1858,7 @@ _rpmalloc_heap_extract_new_span(heap_t* heap, heap_size_class_t* heap_size_class
 
 static void
 _rpmalloc_heap_initialize(heap_t* heap) {
-	memset(heap, 0, sizeof(heap_t));
+	_rpmalloc_memset_const(heap, 0, sizeof(heap_t));
```

Comment:
Do you have any example of the final generated assembly too see the difference? Is it worth the extra code and platform maintenance?

Reply:
Here __builtin_memset_inline
`leaq	-9(%rbp), %rax
	movb	$0, 4(%rax)
	movl	$0, (%rax)
`
vs `memset`

`leaq	-9(%rbp), %rdi
	xorl	%esi, %esi
	movl	$5, %edx
	callq	memset@PLT
	leaq	.L.str.3(%rip), %rdi
	movb	$0, %al`

`__builtin_memcpy_inline`

`leaq	-9(%rbp), %rax
	movq	-24(%rbp), %rcx
	movb	4(%rcx), %dl
	movb	%dl, 4(%rax)
	movl	(%rcx), %ecx
	movl	%ecx, (%rax)
	leaq	.L.str.4(%rip), %rdi
	movb	$0, %al`

vs `memcpy`

`leaq	-9(%rbp), %rdi
	movq	-24(%rbp), %rsi
	movq	-32(%rbp), %rdx
	callq	memcpy@PLT
	xorl	%eax, %eax
	addq	$32, %rsp
	popq	%rbp`

So basically the _inline flavor guarantees no call are generated.